contracts/vaults/Claim.sol

```diff
@@ -0,0 +1,202 @@
+        // to HAT so it could call swapAndSend in a separate tx.
+        asset.safeApprove(address(registry), claimBounty.hackerHatVested + claimBounty.governanceHat);
+        registry.addTokensToSwap(
+            asset,
+            claim.beneficiary,
+            claimBounty.hackerHatVested,
+            claimBounty.governanceHat
+        );
+
+        // make sure to reset approval
+        asset.safeApprove(address(registry), 0);
+
+        // emit event before deleting the claim object, bcause we want to read beneficiary and bountyPercentage
+        emit ApproveClaim(
+            _claimId,
+            msg.sender,
+            claim.beneficiary,
+            claim.bountyPercentage,
+            tokenLock,
+            claimBounty
+        );
+    }
+
+    /**
+    * @notice Dismiss the active claim for a bounty submitted by a committee.
+    * Called either by the arbitrator, or by anyone if the claim is after the challenge timeout period.
+    */
+    function dismissClaim(bytes32 _claimId) external isActiveClaim(_claimId) {
+        Claim memory claim = activeClaim;
+        if (claim.challengedAt == 0) revert OnlyCallableIfChallenged();
+        // solhint-disable-next-line not-rely-on-time
+        if (block.timestamp < claim.challengedAt + challengeTimeOutPeriod && msg.sender != arbitrator)
+            revert OnlyCallableByArbitratorOrAfterChallengeTimeOutPeriod();
+        delete activeClaim;
+
+        emit DismissClaim(_claimId);
+    }
+
+    /**
+    * @dev calculate the specific bounty payout distribution, according to the
+    * predefined bounty split and the current bounty percentage
+    * @param _bountyPercentage The percentage of the vault's funds to be paid
+    * out as bounty
+    * @return claimBounty The bounty distribution for this specific claim
+    */
+    function calcClaimBounty(uint256 _bountyPercentage)
+    public
+    view
+    returns(ClaimBounty memory claimBounty) {
+        uint256 totalSupply = totalAssets();
+        if (totalSupply == 0) revert VaultBalanceIsZero();
+        if (_bountyPercentage > maxBounty)
+            revert BountyPercentageHigherThanMaxBounty();
+        uint256 totalBountyAmount = totalSupply * _bountyPercentage;
+
+        uint256 governanceHat =
+        totalBountyAmount * hatBountySplit.governanceHat
+        / HUNDRED_PERCENT_SQRD;
+        uint256 hackerHatVested =
+        totalBountyAmount * hatBountySplit.hackerHatVested
+        / HUNDRED_PERCENT_SQRD;
+
+        totalBountyAmount -= (governanceHat + hackerHatVested) * HUNDRED_PERCENT;
+
+        claimBounty.governanceHat = governanceHat;
+        claimBounty.hackerHatVested = hackerHatVested;
+        claimBounty.hackerVested =
+        totalBountyAmount * bountySplit.hackerVested
+        / HUNDRED_PERCENT_SQRD;
+        claimBounty.hacker =
+        totalBountyAmount * bountySplit.hacker
+        / HUNDRED_PERCENT_SQRD;
+        claimBounty.committee =
```

Comment:
mb better to just give "the rest" to the comittee so we do not have rounding errors (not that it matters really but it is more precise)

Reply:
https://github.com/hats-finance/hats-contracts/pull/316